### PR TITLE
[dbus] add detailed dbus dump

### PR DESCRIPTION
**Pull request by gjc13**

### CODE REVIEW

File: src/dbus/common/dbus_message_dump.hpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace otbr {
+namespace DBus {
+
+/**
+ * This function dumps a DBus message to the log
+ *
+ * @param[in] aMessage  The DBus message to dump.
+ */
+void DumpDBusMessage(DBusMessage &aMessage);
```
[the remark this answers is not in the document]
Comment (gjc13) [reply src/dbus/common/dbus_message_dump.hpp:42]: Unfortunately, `libdbus` doesn't provide us with APIs manipulating `const DBusMessage`

File: src/dbus/common/dbus_message_dump.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "dbus_message_dump.hpp"
+
+#include <inttypes.h>
+
+#include "common/logging.hpp"
```
[the remark this answers is not in the document]
Comment (gjc13) [reply src/dbus/common/dbus_message_dump.cpp:33]: This file refers to `src/common/logging.hpp`.

File: src/dbus/common/dbus_message_dump.cpp [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    }
+    return;
+}
+
+void DumpDBusMessage(DBusMessage &aMessage)
+{
+    DBusMessageIter    iter;
+    std::ostringstream sout;
+
+    VerifyOrExit(dbus_message_iter_init(&aMessage, &iter), OTBR_NOOP);
```
Comment (bukepo) [inline src/dbus/common/dbus_message_dump.cpp:173]: Should it print error here?

File: src/dbus/common/dbus_message_dump.cpp [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            if (dbus_message_iter_get_arg_type(&subIter) == DBUS_TYPE_VARIANT)
+            {
+                DBusMessageIter valueIter;
+                dbus_message_iter_recurse(&subIter, &valueIter);
+                DumpDBusMessage(sout, &valueIter);
+            }
+            sout << "}, ";
+            break;
+        }
+        case DBUS_TYPE_INVALID:
```
Comment (bukepo) [inline src/dbus/common/dbus_message_dump.cpp:158]: does this indicate an error?

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by Irving-cl — APPROVED** [review]
LGTM👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**